### PR TITLE
GL-446. Update wrapper to do fingerprinting and merging metrics.

### PR DESCRIPTION
**Pull request by gbggrant**

### CODE REVIEW

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.changelog.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# 1.0.0
+2022-01-14 (Date of Last Commit)
+
+* Initial Release of BroadInternalRNAWithUMIs pipeline.
```
Comment (jessicaway) [inline pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.changelog.md:4]: Should we include a short description of the purpose of the pipeline? Something like "The BroadInternalRNAWithUMIs pipeline wraps the RNAWithUMIs pipeline and performs additional steps that rely on Broad specific infrastructure."

File: verification/VerifyBroadInternalRNAWithUMIs.wdl [hunk 2]
```diff
@@ -0,0 +1,48 @@
+version 1.0
```
Comment (jessicaway) [inline verification/VerifyBroadInternalRNAWithUMIs.wdl:1]: Can we just use the VerifyRNAWithUMIs directly? Do we need this file?

**Review by jessicaway — APPROVED** [review]
Couple minor comments and suggestions. Looks good overall

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.wdl [hunk 3]
```diff
@@ -32,11 +39,15 @@ workflow BroadInternalRNAWithUMIsPipeline {
...
   File refFlat = if (reference_build == "hg19") then "gs://broad-gotc-test-storage/rna_seq/hg19/Homo_sapiens_assembly19.refFlat.txt" else "gs://broad-gotc-test-storage/rna_seq/hg38/GRCh38_gencode.v27.refFlat.txt"
+  # TODO - Need to push the hg38 haplotype db up to google bucket.
+  File haplotype_database_file = if (reference_build == "hg19") then "gs://gcp-public-data--broad-references/hg19/v0/Homo_sapiens_assembly19.haplotype_database.txt" else "gs://broad-gotc-test-storage/rna_seq/hg38/Homo_sapiens_assembly38_noALT_noHLA_noDecoy.haplotype_database.txt"
   File ribosomalIntervals = if (reference_build == "hg19") then "gs://broad-gotc-test-storage/rna_seq/hg19/Homo_sapiens_assembly19.rRNA.interval_list" else "gs://broad-gotc-test-storage/rna_seq/hg38/gencode.v26.rRNA.withMT.interval_list"
   File exonBedFile = if (reference_build == "hg19") then "gs://broad-gotc-test-storage/rna_seq/hg19/gencode.v19.hg19.insert_size_intervals_geq1000bp.bed" else "gs://broad-gotc-test-storage/rna_seq/hg38/gencode.v26.GRCh38.insert_size_intervals_geq1000bp.bed"
```
Comment (nikellepetrillo) [inline pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.wdl:45]: @takutosato Are we good to use this hg19 bed file (gencode.v19.hg19.insert_size_intervals_geq1000bp.bed) ?  I know you found it in Francois' directory, but wanted to make sure this is the right bed file to keep using.

**Review by nikellepetrillo — APPROVED** [review]

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 4]
```diff
@@ -17,10 +17,10 @@ task VerifyPipelineInputs {
         String? read_group_name
         String? sequencing_center = "BI"
 
-        String docker = "python:3.7.2"
+        String docker = "us.gcr.io/broad-dsp-gcr-public/base/python:3.9-debian"
         Int cpu = 1
-        Int memory_mb = 1000
-        Int disk_size_gb = 1
+        Int memory_mb = 2000
+        Int disk_size_gb = ceil(size(bam, "GB") + size(r1_fastq,"GB") + size(r2_fastq, "GB")) + 10
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply tasks/broad/RNAWithUMIsTasks.wdl:23]: Yes, we have a test case for this.

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl [hunk 5]
```diff
@@ -165,20 +166,9 @@ workflow RNAWithUMIsPipeline {
...
-			picard_rna_metrics=CollectRNASeqMetrics.rna_metrics,
-			duplicate_metrics=UMIAwareDuplicateMarking.duplicate_metrics,
-			rnaseqc2_metrics=rnaseqc2.metrics,
-			fingerprint_summary_metrics="gs://broad-gotc-test-storage/rna_seq/example.fingerprinting_summary_metrics",
-			output_basename = GetSampleName.sample_name
-	}
-
-
 	output {
+		# TODO - is sample_name really needed to be output here - used in MergeMetrics in wrapper...
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:170]: It was a TODO, but I've TODONE it. hahahhahahahahhaha

File: verification/VerifyBroadInternalRNAWithUMIs.wdl (hunk 2 above)
Comment (gbggrant) [reply verification/VerifyBroadInternalRNAWithUMIs.wdl]: Yup, done

**Comment by gbggrant** [comment]
retest this please

**Comment by gbggrant** [comment]
retest this please

File: tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/tester/RNAWithUMIsTester.scala [hunk 6]
```diff
@@ -8,11 +8,11 @@ import better.files.File
 import org.broadinstitute.dsp.pipelines.batch.WorkflowTest
```
Comment (jessicaway) [inline tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/tester/RNAWithUMIsTester.scala:8]: The error message in the smart tests is strange (none of the arguments to the tes framework appear invalid to me), but you will need to update the name of the pipeline on line 20 in this file

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.changelog.md (hunk 1 above)
Comment (gbggrant) [reply pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.changelog.md]: Hmm.  Did that, but still failing in smart test.

**Comment by gbggrant** [comment]
@kayleemathews @ekiernan this is the PR we were talking about

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md [hunk 7]
```diff
@@ -1,9 +1,5 @@
-# 0.2.0
-2022-01-12 (Date of Last Commit)
+# 1.0.0
+2022-01-14 (Date of Last Commit)
 
-* Work in progress.
+* Initial Release of RNAWithUMIs pipeline.
```
[the remark this answers is not in the document]
Comment (kayleemathews) [reply pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md:4]: How about something like this:

The RNA with UMIs pipeline is an open-source, cloud-optimized workflow created for processing total RNA isolated with the Transcriptome Capture (TCap) method, but can be used to process any bullk RNA-seq data. Overall, the workflow performs UMI correction, aligns reads to the genome, quantifies gene counts, and calculates quality metrics to produces genome- and transcriptome-aligned BAMs, BAIs, and a merged quality metrics file.

What do you think @ekiernan ?

[the remark this answers is not in the document]
Comment (ekiernan) [reply pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md:4]: @kayleemathews Yes, that's perfect. We often take wording from the first paragraph of the overview doc and put into the changelog for the intitial release. It's also helpful to link to the overview doc when it's available.

[the remark this answers is not in the document]
Comment (gbggrant) [reply pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md:4]: Thanks! I've updated the changelog

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.wdl (hunk 3 above)
Comment (takutosato) [reply pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIsPipeline.wdl]: I have not heard from Francois; I will need to do some digging myself. Could you give me a day or two to get that done?

**Pull request merged**